pkg/services/sqlstore/migrations/ualert/channel.go

```diff
@@ -75,245 +110,197 @@ func (m *migration) getNotificationChannelMap() (channelsPerOrg, defaultChannels
-	// Create receiver and route for this rule.
-	if allChannels == nil {
-		return nil
+func (m *migration) createNotifier(c *notificationChannel) (*PostableGrafanaReceiver, error) {
+	if c.Type == "hipchat" || c.Type == "sensu" {
+		m.mg.Logger.Error("alert migration error: discontinued notification channel found", "type", c.Type, "name", c.Name, "uid", c.Uid)
+		return nil, nil
 	}
 
-	channelIDs := extractChannelIDs(da)
-	if len(channelIDs) == 0 {
-		// If there are no channels associated, we skip adding any routes,
-		// receivers or labels to rules so that it goes through the default
-		// route.
-		return nil
+	uid, ok := m.generateChannelUID()
+	if !ok {
+		return nil, errors.New("failed to generate UID for notification channel")
 	}
 
-	recv, route, err := m.makeReceiverAndRoute(rule.UID, rule.OrgID, channelIDs, defaultChannels[rule.OrgID], allChannels[rule.OrgID])
+	settings, decryptedSecureSettings, err := migrateSettingsToSecureSettings(c.Type, c.Settings, c.SecureSettings)
 	if err != nil {
-		return err
+		return nil, err
 	}
 
-	if recv != nil {
-		amConfig.AlertmanagerConfig.Receivers = append(amConfig.AlertmanagerConfig.Receivers, recv)
-	}
-	if route != nil {
-		amConfig.AlertmanagerConfig.Route.Routes = append(amConfig.AlertmanagerConfig.Route.Routes, route)
-	}
-
-	return nil
+	return &PostableGrafanaReceiver{
+		UID:                   uid,
+		Name:                  c.Name,
+		Type:                  c.Type,
+		DisableResolveMessage: c.DisableResolveMessage,
+		Settings:              settings,
+		SecureSettings:        decryptedSecureSettings,
+	}, nil
 }
 
-func (m *migration) makeReceiverAndRoute(ruleUid string, orgID int64, channelUids []interface{}, defaultChannels []*notificationChannel, allChannels map[interface{}]*notificationChannel) (*PostableApiReceiver, *Route, error) {
-	portedChannels := []*PostableGrafanaReceiver{}
-	var receiver *PostableApiReceiver
-
-	addChannel := func(c *notificationChannel) error {
-		if c.Type == "hipchat" || c.Type == "sensu" {
-			m.mg.Logger.Error("alert migration error: discontinued notification channel found", "type", c.Type, "name", c.Name, "uid", c.Uid)
-			return nil
+// Create one receiver for every unique notification channel. If there are multiple existing default notification channels, these will be stored in a separate receiver
+// containing them all to be used only on the root-level route. This is to simplify the structure of the migrated routes  as we are currently limited to one receiver per route,
+// so nesting the root-level route will require a significantly more complex setup.
+func (m *migration) createReceivers(amConfig *PostableUserConfig, allChannels []*notificationChannel, defaultChannels []*notificationChannel) (map[uidOrID]*PostableApiReceiver, error) {
+	receivers := make(map[uidOrID]*PostableApiReceiver)
+	for _, c := range allChannels {
+		notifier, err := m.createNotifier(c)
```

Comment:
In the case of both Sensu and Hipchat, wouldn't this make `notifier` nil thus making an empty receiver? I take it that we want is to not create the receiver at all - isn't it?

Reply:
Good catch! This is fixed in https://github.com/grafana/grafana/pull/47291/commits/d4cae9db266ca7ab7d562660f00ce5238dad9f63 to be the same outcome as the previous migration now (hipchat and sensu channels are dropped). 

However, I'm not 100% sure if this is how we should do it, what are your thoughts on instead creating an empty contact point for each defunct channel and hooking everything up as normal so they can eventually go and convert that channel to something supported without having to also configure all their routes again?